### PR TITLE
WIP: fix accidently quadratic behavior

### PR DESCRIPTION
**Pull request by matklad**
we've recmoputed item map every time previously :man_facepalming: :man_facepalming: :man_facepalming:

### CODE REVIEW

File: crates/ra_hir/src/query_definitions.rs [hunk 1]
```diff
@@ -36,6 +36,8 @@ pub(super) fn fn_scopes(db: &impl HirDatabase, fn_id: FnId) -> Arc<FnScopes> {
 }
 
 pub(super) fn file_items(db: &impl HirDatabase, file_id: FileId) -> Arc<SourceFileItems> {
+    eprintln!("{:?}", file_id);
```
Comment (kjeremy) [inline crates/ra_hir/src/query_definitions.rs]: Did you mean for this to be here?

Comment (matklad) [reply crates/ra_hir/src/query_definitions.rs]: nope, this is still a WIP :)

**Comment by matklad** [comment]
Found the problem with GC: [this bit](https://github.com/rust-analyzer/rust-analyzer/blob/d05790b9478705c47c9de34bfdee08f823e4e92f/crates/ra_hir/src/module/imp.rs#L69) depends (in salsa's sense) on raw syntax trees, and so is constantly changing.

**Pull request merged**